### PR TITLE
Only log on test failure 

### PR DESCRIPTION
**Pull request by nielsenko**
Use a custom `Realm.logger` and `printOnFailure` to only log on test errors.

At the same time increase `logLevel` to `ALL` to get really detailed trace, if a failure do occur

### CODE REVIEW

**Review by desistefanova — APPROVED** [review]
Great, it is a nice improvement in log readability. I have only one small comment.

**Review by desistefanova — COMMENTED** [review]
@nielsenko, @blagoev, @nirinchev  Do you think it could be a security issue to print all the debug and trace information, even though it is not visible for everyone?

File: test/test.dart [hunk 1]
```diff
@@ -388,7 +395,8 @@ Future<void> setupTests(List<String>? args) async {
     });
   });
 
-  await _printPlatformInfo();
+  // Enable this to print platform info, including current PID
+  // await _printPlatformInfo();
```
[the remark this answers is not in the document]
Comment (blagoev) [reply test/test.dart:399]: @nielsenko can we leave this or perhaps we could use the logger to log this as well?

[the remark this answers is not in the document]
Comment (nielsenko) [reply test/test.dart:399]: Yeah, I can leave it, but it does add a lot of trace to a successful run.

**Comment by nielsenko** [comment]
> @nielsenko, @blagoev, @nirinchev Do you think it could be a security issue to print all the debug and trace information, even though it is not visible for everyone?

Even full logging should not reveal secrets. Also, the logs are not available to non-members.

**Review by desistefanova — APPROVED** [review]

**Pull request merged**